addons/hr_expense/static/src/mixins/qrcode.js

```diff
@@ -0,0 +1,46 @@
+/** @odoo-module */
+
+import { useService } from "@web/core/utils/hooks";
+
+const { onMounted, onPatched, useRef } = owl;
+
+export const ExpenseMobileQRCode = {
+    setup() {
+        this._super();
+        this.root = useRef('root');
+        this.actionService = useService('action');
+
+        onMounted(this.bindAppsIcons);
+        onPatched(this.bindAppsIcons);
+    },
+
+    bindAppsIcons() {
+        const apps = this.root.el.querySelectorAll('.o_expense_mobile_app');
+        if (!apps) {
+            return;
+        }
+
+        const handler = this.handleClick.bind(this);
+        for (const app of apps) {
+            app.addEventListener('click', handler);
+        }
+    },
```

Comment:
This is the same as `useEffect` without the cleanup
```suggestion
        useEffect(
            (el) => {
                if (!el) {
                    return;
                }
                const apps = el.querySelectorAll('.o_expense_mobile_app');
                const handler = this.handleClick.bind(this);
                for (const app of apps) {
                    app.addEventListener('click', handler);
                }
                return () => {
                    for (const app of apps) {
                        app.removeEventListener('click', handler);
                    }
                }
            },
            () => [this.root.el],
        )
    },
```

Reply:
`useEffect()` doesn't work for the nocontent helper ¯\_(ツ)_/¯